zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -182,27 +176,51 @@ public OrchestratorZoneResponse createZone(OrchestratorZoneRequest zoneRequest)
         if (!IdentityZoneHolder.isUaa()) {
             throw new AccessDeniedException("Zones can only be created by being authenticated in the default zone.");
         }
-        String name = zoneRequest.getName();
-        String adminClientSecret = zoneRequest.getParameters().getAdminClientSecret();
-
-        String subdomain = zoneRequest.getParameters().getSubdomain();
-        String id = UUID.randomUUID().toString();
-        subdomain = getSubDomain(subdomain, id);
 
-        IdentityZone identityZone = generateIdentityZone(subdomain, name, id);
+        String name = zoneRequest.getName();
+        String importedServiceInstanceGuid =  zoneRequest.getParameters().getImportedServiceInstanceGuid();
+        if(Objects.nonNull(importedServiceInstanceGuid)){
+            logger.info("Importing existing Identity Zone {}",importedServiceInstanceGuid);
+            try{
+                    if(!isZoneAlreadyImported(importedServiceInstanceGuid)){
```

Comment:
If you want to achieve with single query, you can use left join to get the orchestrator zone along with identity zone and you can check whether orchestrator zone is present.

Here are the steps,

1. Get identity zone left join with orchestrator zone
2. If zone is not found DAO will throw not found exception other check the orchestrator zone is available in the code
3. If orchestrator zone is missing and only identity zone is available, you can go and continue the import
4. otherwise you can throw exception, zone already imported.

Reply:
This logic is implemented as suggested.